exporter/collector/config.go

```diff
@@ -33,6 +35,12 @@ type Config struct {
 	TraceConfig TraceConfig `mapstructure:"trace"`
 
 	MetricConfig MetricConfig `mapstructure:"metric"`
+
+	// MapMonitoredResource is not exposed as an option in the configuration, but
+	// can be used by other exporters to extend the functionality of this
+	// exporter. It allows overriding the function used to map otel resource to
+	// monitored resource.
+	MapMonitoredResource func(pdata.Resource) *monitoredrespb.MonitoredResource
```

Comment:
I wonder if putting this in the each of the MetricConfig and TraceConfig would be better. Plus, logging has some inconsistencies with GCM monitored resource

Reply:
moved to metric config, since logging config doesn't exist yet, and trace doesn't do MR yet